src/io.jl

```diff
@@ -95,23 +102,33 @@ function Base.write(io::IO, d::MRCData; compress=:none, unit_vsize=4096)
     data = parent(d)
     fswap = bswapfromh(h.machst)
     unit_vsize = div(unit_vsize, sizeof(T))
+    if isnothing(buffer)
+        buffer = Vector{T}(undef, unit_vsize)
+    end
+    buffer::Vector{T}
+    # If `buffer` was provided as a parameter then `unit_vsize` is redundant and
+    # we must make sure that it matches `buffer`.
+    unit_vsize = length(buffer)
     vlen = length(data)
     vrem = vlen % unit_vsize
     if vrem != 0
-        @inbounds @views sz += write(newio, fswap.(T.(data[1:vrem])))
+        @inbounds @views buffer[1:vrem] .= fswap.(T.(data[1:vrem]))
```

Comment:
I wonder if this would be more efficient:
```suggestion
        @inbounds map!(fswap ∘ T, buffer[1:vrem], data[1:vrem])
```

Reply:
We could also move the `@views` outside of the `if` block to simplify the contents of the block. Same here and in the following `for` loop.

Reply:
A quick benchmark shows that there is no real performance difference:

```julia
julia> using BenchmarkTools

julia> f(x) = x^2
f (generic function with 1 method)

julia> g(x) = sin(x)
g (generic function with 1 method)

julia> function set_via_broadcast!(a, b)
           a .= f.(g.(b))
       end
set_via_broadcast! (generic function with 1 method)

julia> function set_via_map!(a, b)
           map!(f ∘ g, a, b)
       end
set_via_map! (generic function with 1 method)

julia> a = zeros(10_000);

julia> b = rand(10_000);

julia> @btime set_via_broadcast!($a, $b);
  41.276 μs (0 allocations: 0 bytes)

julia> @btime set_via_map!($a, $b);
  44.948 μs (0 allocations: 0 bytes)
```